src/metabase/db/liquibase/h2.clj

```diff
@@ -0,0 +1,50 @@
+(ns metabase.db.liquibase.h2
+  "Custom implementation of the Liquibase H2 adapter that uppercases all identifiers. See #20611 for more details."
+  (:require [metabase.util :as u])
+  (:import liquibase.database.core.H2Database
+           liquibase.database.jvm.JdbcConnection))
+
+(defn- upcase ^String [s]
+  (some-> s u/upper-case-en))
+
+(defn- h2-database* ^H2Database []
+  (proxy [H2Database] []
+    (quoteObject [object-name object-type]
+      (let [^H2Database this this]
+        (proxy-super quoteObject (upcase object-name) object-type)))
+
+    (mustQuoteObjectName [_object-name _object-type]
+      true)))
+
+;; HACK! Create a [[java.lang.Package]] for the proxy class if one does not already exist. This is needed because:
+;;
+;; 1. Liquibase will throw an NPE if the package for the class does not exist -- see
```

Comment:
Would it be worth linking to the liquibase issue that was opened about this?

Reply:
good idea. I added it to the line below